lib/puppet/functions/new.rb

```diff
@@ -991,12 +991,17 @@
 
   def new_instance(scope, t, *args)
     return args[0] if args.size == 1 && !t.is_a?(Puppet::Pops::Types::PInitType) && t.instance?(args[0])
-    result = assert_type(t, new_function_for_type(t, scope).call(scope, *args))
+    result = assert_type(t, new_function_for_type(t).call(scope, *args))
     return block_given? ? yield(result) : result
   end
 
-  def new_function_for_type(t, scope)
-    @new_function_cache ||= Hash.new() {|hsh, key| hsh[key] = key.new_function.new(scope, loader) }
```

Comment:
Seems like these changes overlap with #8323

Reply:
Ah yeah, I pulled that commit so the acceptance test passes (which tests both fixes). I'll drop it before merging.